### PR TITLE
milestone02: add Pokemon set to Trainer and modified tests accordingly

### PR DESCRIPTION
**Pull request by domhanak**
I have modified Trainer entity to store Pokemon that belong to the Trainer.

There is correction for method findTrainersPokemon in TrainerService and the test were modified too to cover these changes.

Please review and merge.

### CODE REVIEW

File: pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java [hunk 1]
```diff
@@ -79,6 +83,14 @@ void addBadge(Badge badge) {
         return Collections.unmodifiableSet(badges);
     }
 
+    public Set<Pokemon> getPokemons() {
+        return Collections.unmodifiableSet(pokemons);
+    }
+
+    public void setPokemons(Set<Pokemon> pokemons) {
```
Comment (pavelkouril) [inline pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java]: You don't want to set whole collection to replace the current one. Just do `addPokemon` method please which will add a new Pokemon to currect Collection. :)

File: pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java [hunk 2]
```diff
@@ -79,6 +83,14 @@ void addBadge(Badge badge) {
         return Collections.unmodifiableSet(badges);
     }
 
+    public Set<Pokemon> getPokemons() {
```
Comment (pavelkouril) [inline pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java]: Pokemon, not Pokemons :P

File: pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java (hunk 1 above)
Comment (domhanak) [inline pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java]: +1

File: pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java (hunk 2 above)
Comment (domhanak) [inline pkmnleague-persistence/src/main/java/cz/fi/muni/pa165/seminar/pkmnleague/domain/Trainer.java]: +1

**Comment by domhanak** [comment]
Rename pokemons collection to pokemon (because plural of Pokemon is Pokemon)
Modified setPokemon method to addPokemon and made it usable with single objects and Collections.

If everything is correct, feel free to merge.

**Pull request merged**